devtools/etdump/buffer_data_sink.h

```diff
@@ -25,14 +25,24 @@ namespace etdump {
 class BufferDataSink : public DataSinkBase {
  public:
   /**
-   * Constructs a BufferDataSink with a given buffer.
+   * Constructs a BufferDataSink with a given span buffer.
    *
    * @param[in] buffer A Span object representing the buffer where data will be
    * stored.
    */
   explicit BufferDataSink(::executorch::runtime::Span<uint8_t> buffer)
       : debug_buffer_(buffer), offset_(0) {}
 
+  /**
```

Comment:
Are there internal users who need this?

Reply:
not right now but introducing this constructor may make our friends life easier and I'd like to make it as a primary entrance in the future

If our user contructs `BufferDataSink` from Span:

```
void* ptr = malloc(2048)
Span<uint8_t> span= Span(ptr, 2048)
BufferDataSink bds(span)
```
However if from the new constructor:


```
void* ptr = malloc(2048)
BufferDataSink bds(ptr, 2048)
```

More straightforward and sturcure, user doesn't need to take care about span or anything else.

Reply:
I think that the Span ctor would work with syntax like
```
void* ptr = malloc(2048);
BufferDataSink bds({ptr, 2048});
```

Reply:
Program syntactically yes, but user need to figure out what's span is, how to construct it, and how to make the logic shorter, which introduces lots of ramp up cost.
Put into user's shoe, they don't need to knwo what span is. A buffer data sink should be directly constructed on the user-provided memory space, and all type transforming should be happened under-the-neath.